Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/PassUtils.h

```diff
@@ -29,6 +29,9 @@ namespace AZ
             //! Retrieves PassData from a PassDescriptor
             const PassData* GetPassData(const PassDescriptor& descriptor);
 
+            //! Retrieves PassData from a PassDescriptor
+            AZStd::shared_ptr<PassData> GetPassDataPtr(const PassDescriptor& descriptor);
```

Comment:
It's kind of confusing about the ownership of m_passTemplate and m_passData inside PassDescriptor. 

Does this function allow others may keep the passData? and also may change the data?